File: src/main/java/dev/sky_lock/pocketlifevehicle/gui/LoreEditor.java
package dev.sky_lock.pocketlifevehicle.gui;

import dev.sky_lock.menu.InventoryMenu;
import dev.sky_lock.menu.SignEditor;
import dev.sky_lock.pocketlifevehicle.PLVehicle;
import org.bukkit.ChatColor;
import org.bukkit.entity.Player;
import org.bukkit.scheduler.BukkitRunnable;

import java.util.Arrays;
import java.util.List;
import java.util.stream.Collectors;

/**
 * @author sky_lock
 */

public class LoreEditor extends SignEditor {

    public LoreEditor(Player player) {
        super(PLVehicle.getInstance(), player, (packet) -> {
            List<String> lore = Arrays.stream(packet.getLines()).filter(line -> line.length() > 0).map(line -> ChatColor.translateAlternateColorCodes('&', line)).collect(Collectors.toList());
            EditSessions.of(player.getUniqueId()).ifPresent(session -> {
                session.setLore(lore);
            });
            new BukkitRunnable() {
                @Override
                public void run() {
                    InventoryMenu inventoryMenu = (InventoryMenu) player.getOpenInventory().getTopInventory().getHolder();
                    inventoryMenu.open(player, ModelMenuIndex.SETTING.value());
                }
            }.runTaskLater(PLVehicle.getInstance(), 1L);
        });
    }
}

File: src/main/kotlin/dev/sky_lock/pocketlifevehicle/gui/LoreEditor.kt
package dev.sky_lock.pocketlifevehicle.gui

import dev.sky_lock.menu.InventoryMenu
import dev.sky_lock.menu.SignEditor
import dev.sky_lock.pocketlifevehicle.VehiclePlugin
import dev.sky_lock.pocketlifevehicle.packet.UpdateSignPacket
import org.bukkit.ChatColor
import org.bukkit.entity.Player
import org.bukkit.scheduler.BukkitRunnable
import java.util.function.Consumer

/**
 * @author sky_lock
 */
class LoreEditor(player: Player) : SignEditor(VehiclePlugin.instance, player, Consumer { packet: UpdateSignPacket ->
    val lore = packet.lines.filter { line: String -> line.isNotEmpty() }.map { line: String -> ChatColor.translateAlternateColorCodes('&', line) }
    EditSessions.of(player.uniqueId).ifPresent { session: ModelOption -> session.lore = lore }
    object : BukkitRunnable() {
        override fun run() {
            InventoryMenu.of(player).ifPresent { menu ->
                menu.open(player, ModelMenuIndex.SETTING.ordinal)
            }
        }
    }.runTaskLater(VehiclePlugin.instance, 1L)
}) 